RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/SiPixelRawToClusterKernel.dev.cc

```diff
@@ -0,0 +1,810 @@
+                                            ws);
+        }
+
+        constexpr auto lastModule = isPhase2 ? 2049u : nMaxModules + 1;
+        cms::alpakatools::for_each_element_in_block_strided(acc, lastModule, 1025u, [&](uint32_t i) {
+          clus_view[i].clusModuleStart() += clus_view[1024].clusModuleStart();
+        });
+        alpaka::syncBlockThreads(acc);
+
+        if constexpr (isPhase2) {
+          cms::alpakatools::for_each_element_in_block_strided(acc, 3073u, 2049u, [&](uint32_t i) {
+            clus_view[i].clusModuleStart() += clus_view[2048].clusModuleStart();
+          });
+          alpaka::syncBlockThreads(acc);
+
+          cms::alpakatools::for_each_element_in_block_strided(acc, nMaxModules + 1, 3073u, [&](uint32_t i) {
+            clus_view[i].clusModuleStart() += clus_view[3072].clusModuleStart();
+          });
+          alpaka::syncBlockThreads(acc);
+        }
+#ifdef GPU_DEBUG
+        ALPAKA_ASSERT_OFFLOAD(0 == clus_view[0].moduleStart());
+        auto c0 = std::min(maxHitsInModule, clus_view[1].clusModuleStart());
+        ALPAKA_ASSERT_OFFLOAD(c0 == clus_view[1].moduleStart());
+        ALPAKA_ASSERT_OFFLOAD(clus_view[1024].moduleStart() >= clus_view[1023].moduleStart());
+        ALPAKA_ASSERT_OFFLOAD(clus_view[1025].moduleStart() >= clus_view[1024].moduleStart());
+        ALPAKA_ASSERT_OFFLOAD(clus_view[nMaxModules].moduleStart() >= clus_view[1025].moduleStart());
+
+        cms::alpakatools::for_each_element_in_block_strided(acc, nMaxModules + 1, [&](uint32_t i) {
+          if (0 != i)
+            ALPAKA_ASSERT_OFFLOAD(clus_view[i].moduleStart() >= clus_view[i - i].moduleStart());
+          // Check BPX2 (1), FP1 (4)
+          constexpr auto bpix2 = TrackerTraits::layerStart[1];
+          constexpr auto fpix1 = TrackerTraits::layerStart[4];
+          if (i == bpix2 || i == fpix1)
+            printf("moduleStart %d %d\n", i, clus_view[i].moduleStart());
+        });
+#endif
+        // avoid overflow
+        constexpr auto MAX_HITS = TrackerTraits::maxNumberOfHits;
+        cms::alpakatools::for_each_element_in_block_strided(acc, nMaxModules + 1, [&](uint32_t i) {
+          if (clus_view[i].clusModuleStart() > MAX_HITS)
+            clus_view[i].clusModuleStart() = MAX_HITS;
+        });
+
+      }  // end of fillHitsModuleStart kernel operator()
+    };   // end of fillHitsModuleStart struct
+
+    // Interface to outside
+    template <typename TrackerTraits>
+    void SiPixelRawToClusterKernel<TrackerTraits>::makePhase1ClustersAsync(
+        const SiPixelClusterThresholds clusterThresholds,
+        const SiPixelMappingSoAConstView &cablingMap,
+        const unsigned char *modToUnp,
+        const SiPixelGainCalibrationForHLTSoAConstView &gains,
+        const WordFedAppender &wordFed,
+        const uint32_t wordCounter,
+        const uint32_t fedCounter,
+        bool useQualityInfo,
+        bool includeErrors,
+        bool debug,
+        Queue &queue) {
```

Comment:
Maybe move the `Queue&` as the first argument, to be more consistent with Alpaka's interfaces?

(also elsewhere)